drivers/adc/adc_stm32.c

```diff
@@ -537,51 +539,68 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 {
 	const struct adc_stm32_cfg *config = dev->config;
 	ADC_TypeDef *adc = (ADC_TypeDef *)config->base;
+	int ret;
 
 #ifdef CONFIG_SOC_SERIES_STM32G4X
 	if (config->has_temp_channel) {
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(adc1), okay)
 		if ((__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_TEMPSENSOR_ADC1) == channel_id)
 		    && (config->base == ADC1)) {
-			adc_stm32_disable(adc);
+			ret = adc_stm32_disable(adc);
 			adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_TEMPSENSOR);
-			k_usleep(LL_ADC_DELAY_TEMPSENSOR_STAB_US);
+			if (ret == 0) {
+				adc_stm32_enable(adc);
+				k_usleep(LL_ADC_DELAY_TEMPSENSOR_STAB_US);
```

Comment:
If the adc is already enabled (returned !=0), setting the adc_stm32_set_common_path is still done and it must wait for the LL_ADC_DELAY_TEMPSENSOR_STAB_US  stabilisation. This is not conditionned by the adc enable or not situation. That would be : 
```
			adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_TEMPSENSOR);
			k_usleep(LL_ADC_DELAY_TEMPSENSOR_STAB_US);
			if (ret == 0) {
				adc_stm32_enable(adc);
			}
```

Reply:
which does not differ from the current code.

Reply:
I reverted changes on the disable function and reordered the call to disable -> k_usleep -> enable